### PR TITLE
Changed TileMap::set_cell alternative_tile default value to 0

### LINKED ISSUE #57833: New TileSet editor: set_tile only writes NULL
### Godot version

4.0 dev

### System information

2017 macbook pro

### Issue description

Godot version
4.0alpha

System information
mac book pro

Issue description
only seems to write NULL into the map.

Steps to reproduce
I have included a project that demonstrates this.

Minimal reproduction project
[Uploading tile_set issue.zip…]()

### Steps to reproduce

run the included project

### Minimal reproduction project

[tile_set issue.zip](https://github.com/godotengine/godot/files/8028984/tile_set.issue.zip)

### PR DESCRIPTION
**Pull request by IgorKordiukiewicz**
- Fixes #57833

### CODE REVIEW

**Comment by vnen** [comment]
You need to change the default value also where the method is bound: https://github.com/godotengine/godot/blob/3f9a6011fea84a7de3308b8bfd7056827b5eae7d/scene/2d/tile_map.cpp#L3625

**Comment by IgorKordiukiewicz** [comment]
Updated

File: doc/classes/TileMap.xml [hunk 1]
```diff
@@ -227,12 +227,12 @@
...
 			<argument index="3" name="atlas_coords" type="Vector2i" default="Vector2i(-1, -1)" />
-			<argument index="4" name="alternative_tile" type="int" default="-1" />
+			<argument index="4" name="alternative_tile" type="int" default="0" />
 			<description>
 				Sets the tile indentifiers for the cell on layer [code]layer[/code] at coordinates [code]coords[/code]. Each tile of the [TileSet] is identified using three parts:
 				- The source identifier [code]source_id[/code] identifies a [TileSetSource] identifier. See [method TileSet.set_source_id],
 				- The atlas coordinates identifier [code]atlas_coords[/code] identifies a tile coordinates in the atlas (if the source is a [TileSetAtlasSource]. For [TileSetScenesCollectionSource] it should be 0),
-				- The alternative tile identifier [code]alternative_tile[/code] identifies a tile alternative the source is a [TileSetAtlasSource], and the scene for a [TileSetScenesCollectionSource].
+				- The alternative tile identifier [code]alternative_tile[/code] identifies a tile alternative the source is a [TileSetAtlasSource], and the scene for a [TileSetScenesCollectionSource]. Passing -1 will clear the tile.
```
Comment (groud) [inline doc/classes/TileMap.xml:235]: I don't think this precision should be here. You have to set all three args to -1, Vector2i(-1,-1) and -1 to clear a tile, not only the last one.

**Comment by groud** [comment]
I am not strictly against the change, but it annoys me that you would have to call `set_cell(coords, -1, Vector2i(-1,-1), -1)` instead of `set_cell(coords)` to erase a tile.

If we want to change that argument default value we should likely add an `erase_cell(coords)` method to keep erasing a cell simple.

**Comment by IgorKordiukiewicz** [comment]
Updated

**Review by groud — APPROVED** [review]
LGTM.

File: doc/classes/TileMap.xml [hunk 2]
```diff
@@ -57,6 +57,14 @@
 				Clears all cells on the given layer.
 			</description>
 		</method>
+		<method name="erase_cell">
+			<return type="void" />
+			<argument index="0" name="layer" type="int" />
+			<argument index="1" name="coords" type="Vector2i" />
+			<description>
+				Erases the cell on layer [code]layer[/code] at coordinates [code]coords[/code]
```
Comment (akien-mga) [inline doc/classes/TileMap.xml:65]: Missing dot at end of line.

**Pull request merged**